test/fetch/request.js

```diff
@@ -268,6 +269,13 @@ test('undefined signal', t => {
   t.end()
 })
 
+test('error on redirect', async t => {
+  const errorCause = await fetch('http://nodejs.org/', { redirect: 'error' }).catch(e => e.cause)
```

Comment:
This shouldn't use an external url, should use a node http server instead (examples can be found in other tests).

Reply:
The other tests all use invalid URLs and I need this one to be a valid one and to redirect somewhere else. I chose the nodejs website but it can be any website really, as long as we put HTTP instead of https. Any suggestions?

Reply:
The other tests don't depend on a valid url to work; the behavior of the nodejs site could change at any time to not redirect, or the CI or site could have connection issues.

```js
const server = createServer((req, res) => {
    res.statusCode = 302
    res.end()
})
// ...
```

There's examples elsewhere of how to setup node http servers elsewhere in the tests.

Reply:
Thanks! I've followed the patterns on other tests and mine passes now with a custom server. Feel free to take a look again!